README.md

```diff
@@ -1,126 +1,241 @@
+Finally, verify Leap was installed correctly.
+```bash
+nodeos --full-version
+```
+You should see a [semantic version](https://semver.org) string print out followed by a `git` commit hash with no errors. For example:
+```
+v3.1.2-0b64f879e3ebe2e4df09d2e62f1fc164cc1125d1
+```
 
-Generally we recommend performing what we refer to as a "pinned build" which ensures the compiler and boost version remain the same between builds of different Leap versions (Leap requires these versions remain the same otherwise its state needs to be repopulated from a portable snapshot).
+## Source Installation
+You can also build and install Leap from source.
+
+### Prerequisites
+Recent Ubuntu LTS releases are the only operating systems that we support. Other Unix derivatives such as macOS are tended to on a best-effort basis and may not be full featured. If you aren't using Ubuntu, please visit [this page](./docs/00_install/01_build-from-source/00_build-unsupported-os.md) to explore your options.
+
+Notable requirements to build are:
+- C++17 compiler and standard library
+- boost 1.67+
+- CMake 3.8+
+- LLVM 7 - 11 - for Linux only
+  - newer versions do not work
+- openssl 1.1+
+- libcurl
+- curl
+- libusb
+- git
+- GMP
+- Python 3
+- zlib
+
+If you don't already have the Leap repo cloned to your computer, [open a terminal](https://itsfoss.com/open-terminal-ubuntu) and navigate to the folder you want the Leap source code to live in.
+```bash
+cd ~/Downloads
+```
+Clone Leap using HTTPS...
+```bash
+git clone --recursive https://github.com/AntelopeIO/leap.git
+```
+...or SSH.
+```bash
+git clone --recursive git@github.com:AntelopeIO/leap.git
+```
+An HTTPS and SSH clone both result in the same thing, a folder named `leap` containg our source code, it doesn't matter which one you use. Navigate into that folder.
+```bash
+cd leap
+```
+Choose which [release tag](https://github.com/AntelopeIO/leap/releases) or [branch](#branches) you would like to build and check it out. If you don't know, use the [latest release](https://github.com/AntelopeIO/leap/releases/latest). For example, if you want to build release 3.1.2, you would check it out using its tag.
+```bash
+git checkout v3.1.2
+```
 
-#### Building Pinned Build Binary Packages
-In the directory `<leap src>/scripts` you will find the two scripts `install_deps.sh` and `pinned_build.sh`. If you haven't installed build dependencies then run `install_deps.sh`. Then run `pinned_build.sh <dependencies directory> <leap build directory> <number of jobs>`.
+Once you are on the branch or tag (release) you want to build, make sure everything is up-to-date.
+```bash
+git pull
```

Comment:
Depending how exhaustive we're trying to be with the instructions (and it seems now, rather exhaustive), then maybe somehow the `git pull` needs to be prior to the `git checkout` so that the latest tags are pulled before doing the checkout should someone being using an existing checkout.

Reply:
Good catch! What about `git fetch`? That should pull down new tags and branches without spending data/time on the current branch if it is not wanted.

Reply:
Added `git fetch --all --tags`. Let me know if there is some reason that won't work.